benchexec/tools/graves-par.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.coveriteam_verifier_validator as coveriteam
+from benchexec.tools.template import UnsupportedFeatureException
+from benchexec.tools.sv_benchmarks_util import get_data_model_from_task, ILP32, LP64
+import re
+
+
+class Tool(coveriteam.Tool):
+    """
+    Tool infor for Graves, a verifier selector based on Graph Neural Networks
+    We inherit Coveriteam's infrastructure.
+    URL: https://github.com/mgerrard/graves-par
+    """
+
+    REQUIRED_PATHS = [
+        "coveriteam",
+        "bin",
+        "lib",
+        "depends",
+        "predict",
+        "predict/build",
+    ]
+
+    def name(self):
+        return "Graves-Par"
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("graves.sh")
+
+    def cmdline(self, executable, options, task, rlimits):
+        """
+        Graves-PAR takes in three arguments: the program looking to be verified,
+        a property file, and the data model. From their it forms its prediction
```

Comment:
This comment does not match the implementation of this method, is this a bug?

Contrary to the comment, the method adds the data model, the property file, and the input file (in this order).

Note: I would say that relaying on a fixed order of positional command-line arguments (e.g. "first parameter is input file, second parameter is property file") is unusual, confusing, and error-prone. I would recommend to use named parameters for such cases.

Reply:
I agree. This is very brittle (and currently wrong). I have fixed the ordering, but we will make it flag based instead of positional in the future.